app/actions/workers/algorithms.test.js

```diff
@@ -0,0 +1,444 @@
+import expect, { spyOn, restoreSpies, createSpy } from 'expect'
+import sinon from 'sinon'
+import * as algorithms from './algorithms'
+// import Client from '../../utils/piazza-client'
+
+let _client, _handlers, _instance
+
+// describe("James' first bf test", function() {
+// 	it('should pass', function() {
+// 		expect(true).toBeTruthy();
+// 	});
+// });
+
+describe('Algorithms Worker', function() {
+
+	beforeEach(function() {
+		spyOn(console, 'debug');
+		spyOn(console, 'error');
+	})
+
+	afterEach(function() {
+		restoreSpies();
+	})
+
+	describe('start()', function() {
+
+		describe('program flow', function() {
+
+			beforeEach(function() {
+				_handlers = {
+					beforeFetch: createSpy().andReturn(false),
+					onFailure: createSpy().andReturn(false),
+					onTerminate: createSpy().andReturn(false),
+					onUpdate: createSpy().andReturn(false),
+					shouldRun: createSpy().andReturn(false)
+					};
+				_client = {
+					getServices: createSpy().andReturn(Promise.resolve([
+						{
+							resourceMetadata: {
+								description: '',
+								name: '',
+								metadata: {
+									Interface: ''
+								}
+							},
+							serviceId: '',
+							url: ''
+						}
+					]))
+				};
+			});
+
+			afterEach(function() {
+				algorithms.terminate()
+				// try {
+				// 	algorithms.terminate()
+				// }
+				// catch(e) {}
+			});
+
+			it('calls work() function.', function() {
+				algorithms.start(_client, 50, _handlers);
+				expect(_handlers.shouldRun).toHaveBeenCalled();
+			});
+
+			it('calls shouldRun() if shouldRun() outputs false.', function() {
+				algorithms.start(_client, 50, _handlers);
+				// Calls
```

Comment:
This comment _does_ provide some degree of visual clustering of like-operations, but you should let the code speak for itself.